### ISSUE TITLE
maybe have a bug

### ISSUE THREAD

**Issue by fangfangzk** [zqhang/AnomalyCLIP#18]
我发现代码中 similarity_map_list里面的4个map，值都相等。原因可能是把中间的feature加入list时没有clone()？
![image](https://github.com/zqhang/AnomalyCLIP/assets/75557988/28cc8bdd-fe51-4243-baec-119f3fc4b2a0)

**Comment by zqhang** [comment]
Hi! Have you checked it for all images or just a single one?

**Comment by zqhang** [comment]
Thank you, I will check it.

**Comment by CHELSEA234** [comment]
@fangfangzk @zqhang 
I have also noticed this point. May I ask if the performance changes after changing to using x[0]?

**Comment by lyy70** [comment]
Yes, I also found this BUG, I regenerated the exception graph of the four layers and found that they were consistent. The main reason lies in out_token.append(x[0]). When I changed it to out_tokens.append(copy.deepcopy(x[0])), the patchtoken of the four layers was different, but the exception detection performance decreased.

**Comment by kiyoonjeong0305** [comment]
I found this bug, too.

**Comment by tommying** [comment]
@lyy70 你好，请问解决了吗，我发现，只要在train.py中调用model.visual.DAPM_replace(DPAM_layer = 20)这句话，也就是说，只要把 attention换成v-v attention，以后的特征就会变得一样。只要是不替换 v-v attention，后面特征就和前面的不一样。
